### PR TITLE
AWS Lambda Integration

### PR DESCRIPTION
**Pull request by amshamah419**
## Status
In Progress

## Related Issues
fixes: demisto/etc#15631

## Description
Was part of DevOp's integrations. Has been productised.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by amshamah419** [comment]
@kirbles19 - The Documentation is here:
https://github.com/demisto/content/blob/AWS-Lambda/Integrations/AWS-Lambda/AWS-Lambda_description.md
https://github.com/demisto/content/blob/AWS-Lambda/Integrations/AWS-Lambda/AWS-Lambda.yml

File: Integrations/AWS-Lambda/AWS-Lambda.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    if 'LogResult' in response:
+        data.update({'LogResult': base64.b64decode(response['LogResult'])})  # type:ignore
+    if 'Payload' in response:
+        data.update({'Payload': response['Payload'].read().decode("utf-8")})  # type:ignore
+    if 'ExecutedVersion' in response:
+        data.update({'ExecutedVersion': response['ExecutedVersion']})  # type:ignore
+    if 'FunctionError' in response:
+        data.update({'FunctionError': response['FunctionError']})
+
+    ec = {'AWS.Lambda.InvokedFunctions': data}
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.py]: please add a dt here to merge responses by function name

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+        packages and layer archives.
+      type: number
+    - contextPath: AWS.Lambda.AccountUsage. FunctionCount
+      description: The number of Lambda functions.
+      type: number
+  dockerimage: demisto/boto3py3:1.0.0.1030
+  isfetch: false
+  runonce: false
+  script: ''
+  type: python
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: subtype is missing

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+      description: Specify a version or alias to get details about a published version
+        of the function.
+      isArray: false
+      name: qualifier
+      required: false
+      secret: false
+    - default: false
+      description: The AWS Region, if not specified the default region will be used
+      isArray: false
+      name: region
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: add region selection here

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+      description: list of tags associated with the function.
+      type: string
+    - contextPath: AWS.Lambda.Functions.Concurrency.ReservedConcurrentExecutions
+      description: The number of concurrent executions reserved for this function.
+      type: string
+  - arguments:
+    - default: false
+      description: The AWS Region, if not specified the default region will be used
+      isArray: false
+      name: region
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: add region selection here

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 5]
```diff
@@ -0,0 +1,574 @@
...
+        aliases that are pointing to the specific Lambda function version, otherwise
+        the API returns all of the aliases created for the Lambda function.
+      isArray: false
+      name: functionVersion
+      required: false
+      secret: false
+    - default: false
+      description: The AWS Region, if not specified the default region will be used
+      isArray: false
+      name: region
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: add region selection here

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 6]
```diff
@@ -0,0 +1,574 @@
...
+        only if the invocation type is RequestResponse.
+      type: string
+    - contextPath: AWS.Lambda.InvokedFunctions.Region
+      description: The AWS Region.
+      type: string
+  - arguments:
+    - default: false
+      description: The AWS Region, if not specified the default region will be used
+      isArray: false
+      name: region
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: add region selection here

File: Integrations/AWS-Lambda/AWS-Lambda.yml [hunk 7]
```diff
@@ -0,0 +1,574 @@
...
+      description: Specify a version or alias to invoke a published version of the
+        function.
+      isArray: false
+      name: qualifier
+      required: false
+      secret: false
+    - default: false
+      description: The AWS Region, if not specified the default region will be used
+      isArray: false
+      name: region
```
Comment (yuvalbenshalom) [inline Integrations/AWS-Lambda/AWS-Lambda.yml]: add region selection here

**Pull request merged**